### ISSUE TITLE
Optional review quizzes upon completion of a project. 

### ISSUE THREAD

**Issue by WebDev955** [freeCodeCamp/freeCodeCamp#54391]
### Is your feature request related to a problem? Please describe.

As an independent learner, I find that it's very easy to jump from one completed project to a new one without stopping to review my notes from the previous project and collect my thoughts. That part of the process is boring vs the guided projects. An optional review would force me to stop for a moment and continue this interactive way of self taught learning. This has been my experience with the JavaScript courses so far, along with the web design courses. 

### Describe the solution you'd like

Upon the completion of each practice project within a course, an **optional quiz/review** should be unlocked for the Camper. This could be presented in a a variety of ways, but I feel that the main point of it needs to simply go over the major takeaways from the project/course. It should not trap the Camper from continuing though through the courses by needing to get a passing grade.

We all learn differently, and I'm personally very stubborn about forcing myself to study notes. I think optional quizzes/guided reviews would be a fantastic way to give Campers a breather to review new material before they move onto the next project vs simply reading over their own notes.

### Describe alternatives you've considered

1.  Quizzes/reviews could be a traditional multiple choice, true/false quiz with mini scenarios to write out code snippets.  Upon completion, a grade is given, results are shown, and any wrong answers are provided with the relevant freeCodeCamp tutorial article or project scenario steps to review. 

2. It could be a simple summary with a list that outlines the important concepts with notes and code examples and include resource links for further reading for the core topics.

**Comment by Sembauke** [comment]
Hey @WebDev955,

Thank you for giving feedback on our new JavaScript Certification. We are always looking for ways to better the learning flow of new users. The issue is that with our new curriculum (Responsive Web design and JavaScript Algorithms and Data Structure certification) there is no great way to "go back" to a previous lesson where X specific thing is learnt. There is a  pull-request open to fix this, but our current focus is elsewhere.

> Quizzes/reviews could be a traditional multiple choice, true/false quiz with mini scenarios to write out code snippets. Upon completion, a grade is given, results are shown, and any wrong answers are provided with the relevant freeCodeCamp tutorial article or project scenario steps to review.

Although, adding a quiz to the end of a practice project seems like a good idea, the fundamental problem is still present. We also have of course, certification project which show your current skill-level in the concept that were taught in the previous projects.

Basically what I am saying is: If we do add a quiz, there is no good way to point the camper (user) to specific step.

**Comment by a2937** [comment]
In addition, steps get jumbled around time to time as we find issues in the curriculum. So even if we add a link to a step, there is no good way right now to point towards the step where that concept was introduced. Now if we started tagging individual steps where certain concepts are discussed (provided these terms gets reserved), this could be a different story.

**Comment by Sembauke** [comment]
We are planning something like this in our upcoming curriculum
